src/vs/workbench/api/node/mainThreadHandlerRegistry.ts

```diff
@@ -0,0 +1,57 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+/**
+ * MainThreadHandlerRegistry maintains a collection of objects by handle.
+ */
+export class MainThreadHandlerRegistry<T> {
+
+	private _nextChildHandle: number = 0;
+	private _registrations: { [handle: number]: { [childHandle: number]: T }; } = Object.create(null);
```

Comment:
I would always prefer `Map`s, but if VS Code uses this style in similar classes it makes sense to stay consistent.